baselines/FedStar/FedStar/server.py

```diff
@@ -0,0 +1,219 @@
+            strategy=self.strategy,
+            grpc_max_message_length=GRPC_MAX_MESSAGE_LENGTH,
+        )
+
+    def get_on_fit_config_fn(self):
+        def fit_config(
+            rounds=self.rounds,
+            epochs=self.epochs,
+            batch_size=self.batch_size,
+            learning_rate=self.lr,
+        ):
+            if self.current_round != 1:
+                print(
+                    "\nTraining round completed in "
+                    + str(round(time.time() - self.round_time, 2))
+                    + " seconds."
+                )
+            print(
+                "\n%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%\n"
+                + "               Server started "
+                + str(self.current_round)
+                + "th round of training.\n"
+                + "%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%\n"
+            )
+            # Update round start time
+            self.round_time = time.time()
+            return {
+                "rounds": str(self.rounds),
+                "c_round": str(self.current_round),
+                "epochs": str(epochs),
+                "batch_size": str(batch_size),
+                "learning_rate": str(learning_rate),
+            }
+
+        return fit_config
+
+    def get_eval_fn(self, ds_test):
+        import tensorflow as tf
+
+        def evaluate(servr_round, weights, configs):
+            loss, acc = 0, 0
+            self.current_round += 1
+            if (self.current_round - 1) % self.evalution_step == 0:
+                model = Network(num_classes=self.num_classes).get_evaluation_network()
+                model.compile(
+                    optimizer=tf.keras.optimizers.Adam(self.lr),
+                    loss=tf.keras.losses.SparseCategoricalCrossentropy(
+                        from_logits=True, name="loss"
+                    ),
+                    metrics=[
+                        tf.keras.metrics.SparseCategoricalAccuracy(name="accuracy")
+                    ],
+                )
+                model.set_weights(weights)
+                loss, acc = model.evaluate(ds_test, verbose=self.verbose)
+                self.final_accuracy = acc
+                # Clear Memory
+                clear_memory()
+            return float(loss), {"accuracy": float(acc)}
+
+        return evaluate
+
+    def get_accuracy(self):
+        return self.final_accuracy
+
+
+def clear_memory():
+    import tensorflow as tf
+
+    gc.collect()
+    tf.keras.backend.clear_session()
+
+
+def set_logger_level():
+    if "flower" in [
+        logging.getLogger(name).__repr__()[8:].split(" ")[0]
+        for name in logging.root.manager.loggerDict
+    ]:
+        logger = logging.getLogger("flower")
+        logger.setLevel(logging.INFO)
+
+
+def set_gpu_limits(gpu_id, gpu_memory):
+    os.environ["CUDA_VISIBLE_DEVICES"] = gpu_id
+    import tensorflow as tf
```

Comment:
needed to import `tf` again?

Reply:
Thanks for the suggestion, I have removed the imports.